src/OrleansCodeGenerator/SerializerGenerator.cs

```diff
@@ -480,13 +477,52 @@ private static ExpressionSyntax GetObjectCreationExpressionSyntax(Type type)
             else
             {
                 // Create an unformatted object.
-                Expression<Func<object>> getUninitializedObject =
+#if DNXCORE50
+                var bindingFlags = SyntaxFactoryExtensions.GetBindingFlagsParenthesizedExpressionSyntax(
+                    SyntaxKind.BitwiseOrExpression,
+                    BindingFlags.Instance,
+                    BindingFlags.NonPublic,
+                    BindingFlags.Public);
+
+                var activatorCreateInstanceParamsCount = 5;
+                ArgumentSyntax[] createInstanceArguments = new ArgumentSyntax[activatorCreateInstanceParamsCount];
+                var nullLiteralExpressionArgument = SF.Argument(SF.LiteralExpression(SyntaxKind.NullLiteralExpression));
+                createInstanceArguments[0] = SF.Argument(SF.TypeOfExpression(typeInfo.GetTypeSyntax()));
+                createInstanceArguments[1] = SF.Argument(bindingFlags);
+                createInstanceArguments[2] = nullLiteralExpressionArgument;
```

Comment:
I find array initialization syntax easier to read and more obvious. I would flip this upside down so that you construct the parameters first, in local variables, and then create `createInstanceArguments` afterwards, using initializer syntax.